### PR TITLE
txemail: add ability to set sender name

### PR DESCRIPTION
**Pull request by bobheadxi**
Adds a new site config field, `email.senderName`, that can be used to configure a more human-friendly name for when receiving emails.

Thread: https://sourcegraph.slack.com/archives/C03JR7S7KRP/p1679517939835699

## Test plan

```
{
  "email.address": "noreply+dev@cloud.sourcegraph.com",
  "email.senderName": "Robert's Personal Sourcegraph",
  "email.smtp": { ... }
}
```

image

Also verified not setting `email.senderName` still works.

### CODE REVIEW

File: internal/txemail/txtypes/types.go [hunk 1]
```diff
@@ -14,7 +14,6 @@ type InternalAPIMessage struct {
 
 // Message describes an email message to be sent.
 type Message struct {
-	FromName   string   // email "From" address proper name
```
Comment (bobheadxi) [inline internal/txemail/txtypes/types.go:17]: No callsites use this and it was not used correctly in the `render` method either

**Review by rafax — APPROVED** [review]

File: internal/txemail/txemail.go [hunk 2]
```diff
@@ -28,10 +28,15 @@ var emailSendCounter = promauto.NewCounterVec(prometheus.CounterOpts{
 }, []string{"success", "email_source"})
 
 // render returns the rendered message contents without sending email.
-func render(message Message) (*email.Email, error) {
+func render(fromAddress, fromName string, message Message) (*email.Email, error) {
+	from := fromAddress
+	if fromName != "" {
+		from = fmt.Sprintf(`"%s" <%s>`, fromName, fromAddress)
+	}
```
Comment (eseliger) [inline internal/txemail/txemail.go:35]: Go has a type `*mail.Address` for this, perhaps we could make use of that. It does this pretty printing including lots of edge-case handling

**Review by eseliger — APPROVED** [review]
nice!

File: internal/txemail/txemail.go (hunk 2 above)
Comment (bobheadxi) [reply internal/txemail/txemail.go:35]: TIL! updated 😁

**Pull request merged**